pkg/machines/scripts/create_machine.sh

```diff
@@ -0,0 +1,117 @@
+#!/bin/sh
+
+set -u
+VM_NAME="$1"
+SOURCE="$2"
+OS="$3"
+MEMORY_SIZE="$4" # in MiB
+STORAGE_SIZE="$5" # in GiB
+START_VM="$6"
+
+endIfFailed(){
+	if [ $1 -ne 0 ]; then
+	    rm -f "$XMLS_FILE"
+        exit $1
+    fi
+}
+
+# prepare virt-install parameters
+COMPARISON=$(awk 'BEGIN{ print "'$STORAGE_SIZE'"<=0 }')
+if [ "$COMPARISON" -eq 1 ]; then
+    # default to no disk if size 0
+    DISK_OPTIONS="none"
+else
+    DISK_OPTIONS="size=$STORAGE_SIZE,format=qcow2"
+fi
+
+GRAPHICS_PARAM="--graphics spice,listen=127.0.0.1 --graphics vnc,listen=127.0.0.1"
+
+if [ "$OS" = "other-os" -o  -z "$OS" ]; then
+    OS="auto"
+fi
+
+FIRST_1_SOURCE="`echo "$SOURCE" | cut -c 1`"
+FIRST_3_SOURCE="`echo "$SOURCE" | cut -c -3`"
+FIRST_4_SOURCE="`echo "$SOURCE" | cut -c -4`"
+
+if [ "$START_VM" = "true" -a "$FIRST_1_SOURCE" = "/" -a ! -f "$SOURCE" ]; then
+    echo "$SOURCE does not exist or is not a file" 1>&2
+    exit 1
+fi
+
+if [ "$START_VM" = "true" -a \( "$FIRST_1_SOURCE" = "/" -o "$FIRST_4_SOURCE" = "http" -o "$FIRST_3_SOURCE" = "ftp" -o "$FIRST_3_SOURCE" = "nfs" \) ]; then
+    LOCATION_PARAM="--cdrom $SOURCE"
+else
+    # prevents creating duplicate cdroms if start vm is false
+    # or if no source received
+    LOCATION_PARAM=""
```

Comment:
I think the created VM should always get the installation source attached, even if `$START_VM` is false.  Otherwise the created VM is useless.

Reply:
we need to undefine vm in post-install flow and create it again. Url location have special behaviour in virt-install. So we need to pass it again to virt-install. It is stored in metadata and is not useful until we install the vm.